mission_control/test/mission_interface.py

```diff
@@ -102,10 +121,11 @@ def execute_mission(self, mission_id=1):
         self.simulated_mission_control_execute_mission_pub.publish(
             mission_to_execute)
 
-    def wait_for(self, predicate, period=1):
-        while not rospy.is_shutdown():
-            result = predicate()
-            if result:
-                return result
-            rospy.sleep(period)
-        return predicate()
+    def read_behavior_parameters(self, mission_behavior):
```

Comment:
@patapufate meta: this is OK for test, but I'll point out that it is a bit inflexible if there's more than one behavior of the same kind in the mission definition.